Shared/ViewModels/ItemEditorViewModel/RefreshMetadataViewModel.swift

```diff
@@ -0,0 +1,171 @@
+                        self.state = .content
+                        self.eventSubject.send(.refreshTriggered(Date()))
+                    }
+
+                    try await self.refreshMetadata(
+                        metadataRefreshMode: metadataRefreshMode,
+                        imageRefreshMode: imageRefreshMode,
+                        replaceMetadata: replaceMetadata,
+                        replaceImages: replaceImages
+                    )
+
+                    await MainActor.run {
+                        self.state = .refreshing
+                        self.eventSubject.send(.refreshTriggered(Date()))
+                    }
+
+                    try await self.refreshItem()
+
+                    await MainActor.run {
+                        self.state = .content
+                    }
+
+                } catch {
+                    guard !Task.isCancelled else { return }
+
+                    let apiError = JellyfinAPIError(error.localizedDescription)
+                    await MainActor.run {
+                        self.state = .error(apiError)
+                        self.eventSubject.send(.error(apiError))
+                    }
+                }
+            }
+            .asAnyCancellable()
+
+            return .refreshing
+        }
+    }
+
+    // MARK: Metadata Refresh Logic
+
+    private func refreshMetadata(
+        metadataRefreshMode: MetadataRefreshMode,
+        imageRefreshMode: MetadataRefreshMode,
+        replaceMetadata: Bool = false,
+        replaceImages: Bool = false
+    ) async throws {
+        guard let itemId = item.id else { return }
+
+        var parameters = Paths.RefreshItemParameters()
+        parameters.metadataRefreshMode = metadataRefreshMode
+        parameters.imageRefreshMode = imageRefreshMode
+        parameters.isReplaceAllMetadata = replaceMetadata
+        parameters.isReplaceAllImages = replaceImages
+
+        let request = Paths.refreshItem(
+            itemID: itemId,
+            parameters: parameters
+        )
+        _ = try await userSession.client.send(request)
+    }
+
+    // MARK: Refresh Item After Request Queued
+
+    private func refreshItem() async throws {
+        guard let itemId = item.id else { return }
+
+        let totalDuration: Double = 5.0
+        let interval: Double = 0.05
+        let steps = Int(totalDuration / interval)
+
+        // Update progress every 0.05 seconds. Ticks up "1%" at a time.
+        for i in 1 ... steps {
+            try await Task.sleep(nanoseconds: UInt64(interval * 1_000_000_000))
+
+            let currentProgress = Double(i) / Double(steps)
+            await MainActor.run {
+                self.progress = currentProgress
+            }
+        }
+
+        // After waiting for 5 seconds, fetch the updated item
+        let request = Paths.getItem(userID: userSession.user.id, itemID: itemId)
+        let response = try await userSession.client.send(request)
```

Comment:
We should send a `itemMetadataDidChange` notification which the item views listen for and redraw their views by manipulating an `.id(...)`.

We would also have to replace the item in libraries. We will also have to test when you have a series item open in one tab, and update an episode in another tab, and how that updates correctly.